Rx433.cpp

```diff
@@ -65,17 +65,17 @@ void ICACHE_RAM_ATTR rxISR() {
   static int last_state = 0;
   static std::vector<rx433::Pulse> pulse_stream;
   int now = micros();
-  int state = digitalRead(rxPin_);
-  int delta_us = now - last_changed;
+  // Need to invert as p represents state before transition.
+  Pulse p = {now, now - last_changed, digitalRead(rxPin_)};
 
   // Ignore very short glitches that we missed.
-  if (state == last_state) return;
-  last_state = state;
+  if (p.state == last_state) return;
+  last_state = p.state;
 
   // Streamed glich filter
   // Ignore any pulse (1 or 0) shorter than kGlitchUs.
-  if (!pulse_stream.empty() && delta_us < kGlitchUs) {
-    last_changed -= pulse_stream.back().delta_us;
+  if (!pulse_stream.empty() && p.delta_us < kGlitchUs) {
+    last_changed = pulse_stream.back().time_us;
     pulse_stream.pop_back();
```

Comment:
These lines needs to be reversed, as last_changed should contain the time_us of the last element in the stream after the last one was dropped.

Reply:
That will unfortunately not work when there are exactly 1 pulse in the list already. Reverting to use delta_ms instead to also support that case.

Actually, I think time_us is never really used so could be removed fully from the Pulse struct.